bson/json_util.py

```diff
@@ -639,6 +639,9 @@ def _parse_canonical_regex(doc):
     if len(regex) != 2:
         raise TypeError('Bad $regularExpression must include only "pattern"'
                         'and "options" components: %s' % (doc,))
+    if not isinstance(regex['options'], str):
+        raise TypeError('Bad $regularExpression options, options must be '
+                        'string, was type %s' % (type(regex['options'])))
     return Regex(regex['pattern'], regex['options'])
```

Comment:
I suggest refactoring to `regex['options']` to a local variable to avoid looking it up repeatedly and improve readability.

Reply:
Done.